quasiquotes3/shared/src/main/scala/scala/meta/quasiquotes/Api.scala

```diff
@@ -0,0 +1,196 @@
+package scala.meta
+package quasiquotes
+
+import scala.meta.internal.trees.quasiquote
+import scala.meta.internal.quasiquotes._
+import scala.meta.parsers.Parse
+import scala.meta._
+
+type QuasiquoteUnapply
+
+private[meta] trait Api {
+
+  // apply methods
+  extension (inline sc: StringContext) {
+    transparent inline def q(inline args: Any*): Tree = ${ ReificationMacros.termImpl('sc, 'args) }
+    transparent inline def param(inline args: Any*): Term.Param = ${ ReificationMacros.termParamImpl('sc, 'args) }
+    transparent inline def t(inline args: Any*): Type = ${ ReificationMacros.typeImpl('sc, 'args) }
+    transparent inline def tparam(inline args: Any*): Type.Param = ${ ReificationMacros.typeParamImpl('sc, 'args) }
+    transparent inline def p(inline args: Any*): Tree = ${ ReificationMacros.caseOrPatternImpl('sc, 'args) }
+    transparent inline def init(inline args: Any*): Init = ${ ReificationMacros.initImpl('sc, 'args) }
+    transparent inline def self(inline args: Any*): Self = ${ ReificationMacros.selfImpl('sc, 'args) }
+    transparent inline def template(inline args: Any*): Template = ${ ReificationMacros.templateImpl('sc, 'args) }
+    transparent inline def mod(inline args: Any*): Mod = ${ ReificationMacros.modImpl('sc, 'args) }
+    transparent inline def enumerator(inline args: Any*): Enumerator = ${ ReificationMacros.enumeratorImpl('sc, 'args) }
+    transparent inline def importer(inline args: Any*): Importer = ${ ReificationMacros.importerImpl('sc, 'args) }
+    transparent inline def importee(inline args: Any*): Importee = ${ ReificationMacros.importeeImpl('sc, 'args) }
+    transparent inline def source(inline args: Any*): Source = ${ ReificationMacros.sourceImpl('sc, 'args) }
+  }
+
+  // unapply methods
+  extension (stringContext: scala.StringContext)
```

Comment:
Why do we need to repeat the methods here?

Reply:
they have different names, and those names point to different parsers in `ReificationMacros`